domains/authorization/Authorization.API/API/Authorization/[Features]/CreateClientCommand.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Data;
+using API.Models;
+using API.Repository;
+using API.ValueObjects;
+using MediatR;
+
+namespace API.Authorization._Features_;
+
+public record CreateClientCommandHandler(IUnitOfWork UnitOfWork, IClientRepository ClientRepository) : IRequestHandler<CreateClientCommand, CreateClientCommandResult>
+{
+    public async Task<CreateClientCommandResult> Handle(CreateClientCommand request, CancellationToken cancellationToken)
+    {
+        await UnitOfWork.BeginTransactionAsync();
+        var client = Client.Create(request.IdpClientId, request.Name, request.ClientType, request.RedirectUrl);
+        await ClientRepository.AddAsync(client, cancellationToken);
+        await UnitOfWork.CommitAsync();
+
+        return new CreateClientCommandResult(client.Id, client.IdpClientId, client.Name, client.ClientType, client.RedirectUrl);
+    }
+}
+
+public record CreateClientCommand(IdpClientId IdpClientId, ClientName Name, ClientType ClientType, string RedirectUrl) : IRequest<CreateClientCommandResult>;
+
+public record CreateClientCommandResult(Guid Id, IdpClientId IdpClientId, ClientName Name, ClientType ClientType, string RedirectUrl);
```

Comment:
Does an internal client have a redirect Url as well?

Reply:
Nope. Not as it is right now. But this is mainly tool for creating external clients. We can make field nullable.